### PR TITLE
Cluster of fixes

### LINKED ISSUE #3661: Calendar Design Overhaul
This is something that should be done now as it really needs to be updated. In the calendar currently implemented, there are numerous bugs and things that are problematic.

### Current Calendar System Issues
- The design is very unintuitive. To remove an IN block you have to click on the specific day and try to mash on the block to edit it.
- An IN requires an OUT for every time a doctor is in office. We should be using the duration that the person sets as the actual time and get rid of the OUT block.
- You cannot drag a calendar appointment to another day. You have to manually go in and change the date.
- The duration is measured in minutes for each event when it should be showing two time fields (e.g. 4:30pm to 5:30pm with military time support if needed).
- Recurring events are very finicky and sometimes outright disappear if you try deleting one in the series.
- The small month view in the left hand corner is too small.

### Proposed Solution
I suggest we either build off our old system and solve the bugs or create an entirely new calendar system. Since we want doctors to not have issues working with the calendar, this is a pretty high priority.

### LINKED ISSUE #3699: Layout editor's "ID" and "Label" fields are too small to comfortably use
Administration >Layouts' "ID" and "Label" fields are too small to work with comfortably, especially ID where you need to be able to see the whole of the text. For me, ID is 6 chars long, and Label 16 chars long. ID is prone to errors (e.g. inserting a space); uncomfortable when you need to work with it closely trying to cram and hone wording into the very minimal maximum spacing; and difficult to scan across the page with the eye when looking for text. Compare the with the Description field which gets a much bigger box and is optional, where as ID is required.

It takes considerably longer to write a form, working against these issues, than it would if there were more space to work in.

### PR DESCRIPTION
**Pull request by tywrenn**
#### Short description of what this resolves:
- Updates Nation Notes to use BS4 grid columns
- Fixes #3699
- Fixes #3661
- Updates admin Layouts page to a more bs4-friendly page
- Fixes setting the font size in globals. It doesn't look pretty. In my opinion this feature should be removed.
- Fixes some JS lint-related issues to prepare for the big addition of ESLint.
- Other fixes

### CODE REVIEW

**Comment by tywrenn** [comment]
This is a lot of fixing on my part yet one of the fixes I did was redesign the calendar a bit...and they said it couldn't be done xD
image

**Comment by tywrenn** [comment]
This is now gonna overhaul the calendar system on top of what I also fixed.

**Comment by tywrenn** [comment]
Calendar areas that will be overhauled in this pull:
- The persistent month area view will be blown up bigger as well as the sidebar
- The top bar for the calendar will be converted to a flex grid-based layout
- The collapse button for the sidebar will actually look like a collapse button
- Duration in the add/edit event screen will be converted to time (and converted to use datetimepicker)
- All tables in the view will be converted to BS4 tables
- Events will now have a border-radius added to them for a more modernized feel
- The default colors will be changed for event categories (This will not be reflective on migrating from 5.0.2 due to customization possibly being lost. To get the new colors a fresh install is a must.)
- An OUT block will be added whenever you create an IN block (this will be able to be turned off in globals)
- Overall this design will target iPad usage and mobile devices and will enhance usage on touch-based devices.

Things that will need to be solved in follow-up PR's:
- Recurring events are very finicky and sometimes outright disappear if you try deleting one in the series.
- You cannot drag a calendar appointment to another day. You have to manually go in and change the date. (This is something that will be difficult since you have to create logic that detects what day you dragged it to)
- To remove an IN block you have to click on the specific day and try to mash on the block to edit it. (This requires JS experience which I don't have)

**Comment by tywrenn** [comment]
@bradymiller Ready for testing and merging

File: sql/database.sql [hunk 1]
```diff
@@ -6801,21 +6801,21 @@ CREATE TABLE `openemr_postcalendar_categories` (
...
+INSERT INTO `openemr_postcalendar_categories` VALUES (13,'preventive_care_services', 'Preventive Care Services', '#d3c6ec', 'Preventive Care Services', 0, NULL, 'a:5:{s:17:"event_repeat_freq";s:1:"0";s:22:"event_repeat_freq_type";s:1:"0";s:19:"event_repeat_on_num";s:1:"1";s:19:"event_repeat_on_day";s:1:"0";s:20:"event_repeat_on_freq";s:1:"0";}', 0, 900, 0, 0, 0, 0, 0,0,1,13,'encounters|notes');
+INSERT INTO `openemr_postcalendar_categories` VALUES (14,'ophthalmological_services', 'Ophthalmological Services', '#febe89', 'Ophthalmological Services', 0, NULL, 'a:5:{s:17:"event_repeat_freq";s:1:"0";s:22:"event_repeat_freq_type";s:1:"0";s:19:"event_repeat_on_num";s:1:"1";s:19:"event_repeat_on_day";s:1:"0";s:20:"event_repeat_on_freq";s:1:"0";}', 0, 900, 0, 0, 0, 0, 0,0,1,14,'encounters|notes');
+INSERT INTO `openemr_postcalendar_categories` VALUES (15,'group_therapy', 'Group Therapy' , '#adb5bd' , 'Group Therapy', 0, NULL, 'a:5:{s:17:"event_repeat_freq";s:1:"0";s:22:"event_repeat_freq_type";s:1:"0";s:19:"event_repeat_on_num";s:1:"1";s:19:"event_repeat_on_day";s:1:"0";s:20:"event_repeat_on_freq";s:1:"0";}', 0, 3600, 0, 0, 0, 0, 0, 3, 1, 15,'encounters|notes');
```
Comment (bradymiller) [inline sql/database.sql:6818]: TODO for me:
Gonna need some stuff in sql upgrade to fix this for upgraders

File: library/globals.inc.php [hunk 2]
```diff
@@ -231,11 +231,11 @@ function gblTimeZones()
...
-                '0.625rem' => '10px',
-                '0.75rem' => '12px',
-                '0.875rem' => '14px',
-                '1rem' => '16px',
-                '1.125rem' => '18px',
+                '0.625' => '10px',
+                '0.75' => '12px',
+                '0.875' => '14px',
+                '1.0' => '16px',
+                '1.125' => '18px',
```
Comment (bradymiller) [inline library/globals.inc.php:238]: TODO for me:
Gonna need some stuff in sql upgrade to fix this for upgraders

Comment (bradymiller) [reply library/globals.inc.php:238]: this should go into sql/5_0_2-to-6_0_0_upgrade.sql:
```
UPDATE `globals` SET `gl_value`='0.625' WHERE `gl_name`='font-size' AND `gl_value`='0.625rem';
UPDATE `globals` SET `gl_value`='0.75' WHERE `gl_name`='font-size' AND `gl_value`='0.75rem';
UPDATE `globals` SET `gl_value`='0.875' WHERE `gl_name`='font-size' AND `gl_value`='0.875rem';
UPDATE `globals` SET `gl_value`='1.0' WHERE `gl_name`='font-size' AND `gl_value`='1rem';
UPDATE `globals` SET `gl_value`='1.125' WHERE `gl_name`='font-size' AND `gl_value`='1.125rem';
```

File: sql/database.sql (hunk 1 above)
Comment (tywrenn) [reply sql/database.sql:6818]: This one's kinda difficult cause some people have removed some categories and some have renamed some.

File: library/globals.inc.php (hunk 2 above)
Comment (tywrenn) [reply library/globals.inc.php:238]: Done

File: sql/5_0_2-to-6_0_0_upgrade.sql [hunk 3]
```diff
@@ -399,6 +399,12 @@ ALTER TABLE `users_secure` ADD `password_history4` varchar(255);
 
 UPDATE `globals` SET `gl_value`=3 WHERE `gl_name`='password_history' AND `gl_value`=1;
 
+UPDATE `globals` SET `gl_value`='0.625' WHERE `gl_name`='font-size' AND `gl_value`='0.625rem';
+UPDATE `globals` SET `gl_value`='0.75' WHERE `gl_name`='font-size' AND `gl_value`='0.75rem';
+UPDATE `globals` SET `gl_value`='0.875' WHERE `gl_name`='font-size' AND `gl_value`='0.875rem';
+UPDATE `globals` SET `gl_value`='1.0' WHERE `gl_name`='font-size' AND `gl_value`='1rem';
+UPDATE `globals` SET `gl_value`='1.125' WHERE `gl_name`='font-size' AND `gl_value`='1.125rem';
```
Comment (bradymiller) [inline sql/5_0_2-to-6_0_0_upgrade.sql:406]: new items go to the bottom of this script (this is a general rule except for weird cases)

File: sql/database.sql (hunk 1 above)
Comment (bradymiller) [reply sql/database.sql:6818]: next installment :)
agree a bit limited on this. in this case, just gonna change entries that are still at default setting:
```
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#dee2e6' WHERE `pc_constant_id`='no_show' AND `pc_catcolor`='#DDDDDD';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#cce5ff' WHERE `pc_constant_id`='in_office' AND `pc_catcolor`='#99CCFF';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#fdb172' WHERE `pc_constant_id`='out_of_office' AND `pc_catcolor`='#99FFFF';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#e9ecef' WHERE `pc_constant_id`='vacation' AND `pc_catcolor`='#EFEFEF';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#ffecb4' WHERE `pc_constant_id`='office_visit' AND `pc_catcolor`='#FFFFCC';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#8663ba' WHERE `pc_constant_id`='holidays' AND `pc_catcolor`='#9676DB';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#2374ab' WHERE `pc_constant_id`='closed' AND `pc_catcolor`='#2374AB';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#ffd351' WHERE `pc_constant_id`='lunch' AND `pc_catcolor`='#FFFF33';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#93d3a2' WHERE `pc_constant_id`='established_patient' AND `pc_catcolor`='#CCFF33';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#a2d9e2' WHERE `pc_constant_id`='new_patient' AND `pc_catcolor`='#CCFFFF';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#b02a37' WHERE `pc_constant_id`='reserved' AND `pc_catcolor`='#FF7777';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#ced4da' WHERE `pc_constant_id`='health_and_behavioral_assessment' AND `pc_catcolor`='#C7C7C7';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#d3c6ec' WHERE `pc_constant_id`='preventive_care_services' AND `pc_catcolor`='#CCCCFF';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#febe89' WHERE `pc_constant_id`='ophthalmological_services' AND `pc_catcolor`='#F89219';
UPDATE `openemr_postcalendar_categories` SET `pc_catcolor`='#adb5bd' WHERE `pc_constant_id`='group_therapy' AND `pc_catcolor`='#BFBFBF';
```

Comment (bradymiller) [reply sql/database.sql:6818]: as with other, this can go to bottom of sql/5_0_2-to-6_0_0_upgrade.sql script

**Comment by tywrenn** [comment]
@bradymiller Done :)

**Comment by bradymiller** [comment]
@tywrenn , bringing in. code looks good and testing looks awesome! thanks for yet another batch of great improvements! -brady

**Pull request merged**